gsdk/tests/rpc.rs

```diff
@@ -355,6 +359,52 @@ async fn test_program_counters() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn test_calculate_reply_for_handle() -> Result<()> {
+    let node = dev_node();
+
+    let salt = vec![];
+    let pid = ProgramId::generate_from_user(CodeId::generate(demo_ping::WASM_BINARY), &salt);
+
+    // 1. upload program.
+    let signer = Api::new(Some(&node_uri(&node)))
+        .await?
+        .signer("//Alice", None)?;
+
+    signer
+        .calls
+        .upload_program(
+            demo_ping::WASM_BINARY.to_vec(),
```

Comment:
use a demo which includes state, you can add a new method in `demo-messenger`, it is just for testing the clients, for example a `counter`

```rust
fn method_change_state_or_other_names() -> i32 {
   *state += 1;
   return state
}
```

1. *simulate the call and get the updated state
2. (advanced: yes, this requires demo-meta ) query the state and it has not changed
3. *run the call check the result is the same

Reply:
After closer look
- there are no methods in `demo-new-meta` that change the state
- exists example `demo-mul-by-const`, but it is not referenced

Just updated test to use `demo-new-meta`, obviously, the state doesn't change

Reply:
sorry my bad, ur right, I just found that both of the two don't have state mutations ))

only `new-meta` has metadata functions among our examples but it relates to too many tests that modify it is not the ideal way.

I'd suggest you to use `demo-mul-by-const` to verify the state changes since the result of the simulation matches the real execution result is the thing to test

btw, like what you said, this RPC calls obviously doesn't change states, and it is how it works since it is just a read call without signature and will never enter the transaction pool, so skipping verify-state-not-changes makes sense as well!

Reply:
actually, `demo_mul_by_const` does not change state in the handle(

Reply:
addув test to demo-vec and state change, review pls